Create rule S6566: Use DateTimeOffset instead of DateTime (#1717)
SonarSource/rspec: rules/S6566/howToFix.adoc

== How to fix it

> In most cases, you can directly replace `DateTime` with `DateTimeOffset`.
> When hardcoding dates with local kind keep in mind that the offset is timezone dependent so it should be set according to which timezone that data is representing.

Reviewer: ```suggestion
When hardcoding dates with local kind keep in mind that the offset is timezone dependent so it should be set according to which timezone that data is representing.
```
I don't understand what this means - could you maybe me rewrite it a bit clearer?

Reviewer 2: This was the result of a long discussion with @csaba-sagi-sonarsource about [this comment](https://github.com/SonarSource/rspec/pull/1717#discussion_r1158904524). 
The goal was to raise awareness when migrating a hardcoded date to `DateTimeOffset`:
`DateTimeOffset myDate = new DateTimeOffset(2008, 6, 19, 7, 0, 0, TimeSpan.FromHours(-7));`
since you need to consider the offset based on DST/timezone on that particular date/place.
If it's not clear enough I can improve the sentence, let me know.

> When hardcoding dates with local kind keep in mind that the offset is timezone dependent so it should be set according to which timezone that data is representing.

Reviewer 3: ```suggestion
When hardcoding dates with local kind, remember that the offset is timezone dependent, so it should be set according to which timezone that data represents.
```

> For more information refer to DateTime and DateTimeOffset documentation from Microsoft (see "Resources" section below).

Reviewer 3: ```suggestion
For more information, refer to DateTime and DateTimeOffset documentation from Microsoft (see the "Resources" section below).
```
